client/core/trade.go

```diff
@@ -1270,6 +1276,90 @@ func (t *trackedTrade) isSwappable(ctx context.Context, match *matchTracker) (re
 	return false, false
 }
 
+// updateDynamicSwapOrRedemptionFeesPaid updates the fees used for dynamic fee
+// checker transactions. We do not know the exact fees the tx will use until
+// they are mined, so this waits until they are mined and updates the value for
+// the entire trade.
+//
+// TODO: This method does not persist if the client is shut down. It probably
+// should because payed fees will be incorrect forever for this trade if not.
+func (t *trackedTrade) updateDynamicSwapOrRedemptionFeesPaid(ctx context.Context, match *matchTracker, isInit bool) {
+	wallet := t.wallets.fromWallet
+	if !isInit {
+		wallet = t.wallets.toWallet
+	}
+	feeChecker, dynamic := wallet.Wallet.(asset.DynamicSwapOrRedemptionFeeChecker)
+	if !dynamic {
+		return
+	}
+	stopChecks := func() {
+		if isInit {
+			match.needsDynamicSwapFeeConfirms = false
+		} else {
+			match.needsDynamicRedemptionFeeConfirms = false
+		}
+	}
+	txType := "swap"
+	if !isInit {
+		txType = "redemption"
+	}
+	var coinID []byte
+	// Check if a swap or redeem coin id has been populated in the
+	// match tracker. If it has we ask the wallet for the fees paid
+	// and add that to either the total swap or redeem fees for the
+	// trade.
+	if isInit {
+		coinID = []byte(match.MetaData.Proof.MakerSwap)
+		if match.Side != order.Maker {
+			coinID = []byte(match.MetaData.Proof.TakerSwap)
+		}
+	} else {
+		coinID = []byte(match.MetaData.Proof.MakerRedeem)
+		if match.Side != order.Maker {
+			coinID = []byte(match.MetaData.Proof.TakerRedeem)
+		}
+	}
+	contractData := match.MetaData.Proof.ContractData
```

Comment:
Seems to need `CounterContract` instead for the redeem.  Otherwise this would be e.g. your BTC contract's redeem script.

Reply:
Oh yeah... randomly didn't hit this testing... fixed: https://github.com/decred/dcrdex/compare/56872cb137011870f79609b4d065794b8035f930..e04a5d3ef44253041f2a1f81a64caf4dbdfa1bbf